### PR TITLE
Use `canAccess` to simplify device api key permissions

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

### CODE REVIEW

File: src/lib/auth.ts [hunk 1]
```diff
@@ -117,8 +140,8 @@ export const DEVICE_API_KEY_PERMISSIONS = [
 
 ROLES['device-api-key'] = [
 	...DEVICE_API_KEY_PERMISSIONS,
-	`resin.user.get`,
-	`resin.user__has__public_key.get`,
+	`resin.user.read`,
+	`resin.user__has__public_key.read`,
```
Comment (Page-) [inline src/lib/auth.ts]: Not technically necessary, but standardizing on `.read` is nice, and made my grepping far easier ;)

File: src/lib/auth.ts [hunk 2]
```diff
@@ -57,58 +63,75 @@ export const ROLES: {
 };
 
 export const DEVICE_API_KEY_PERMISSIONS = [
-	`resin.device_type.get?describes_device/any(d:d/${matchesActor})`,
-	`resin.device.get?${matchesActor}`,
+	`resin.device_type.read?describes_device/canAccess()`,
```
Comment (Page-) [inline src/lib/auth.ts]: This technically expands the permissions to also be able to read device types of dependent devices, but I think that's actually correct

File: src/lib/auth.ts [hunk 3]
```diff
@@ -57,58 +63,75 @@ export const ROLES: {
...
+	`resin.device.read?${matchesActor}`,
 	`resin.device.update?${matchesActor}`,
-	`resin.supervisor_release.get?should_manage__device/any(d:d/${matchesActor})`,
-	`resin.application.read?${ownsDevice} or depends_on__application/any(da:da/${ownsDevice})`,
-	`resin.application_tag.read?application/any(a:a/${ownsDevice} or a/depends_on__application/any(da:da/${ownsDevice}))`,
-	`resin.device_config_variable.get?device/any(d:d/${matchesActor} or d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
+	`resin.supervisor_release.read?should_manage__device/canAccess()`,
```
Comment (Page-) [inline src/lib/auth.ts]: This technically expands the permissions to also be able to read supervisor releases of dependent devices, which is a bit redundant but I think it's the right thing to do

File: src/lib/auth.ts [hunk 4]
```diff
@@ -57,58 +63,75 @@ export const ROLES: {
...
+	`resin.device_tag.read?device/canAccess()`,
 	`resin.device_tag.set?device/any(d:d/${matchesActor})`,
-	`resin.application_config_variable.get?${applicationControlsDevice}`,
-	`resin.release.read?should_be_running_on__device/any(d:d/${matchesActor}) or ${belongsToApplicationOrIsManagedByDevice}`,
-	`resin.release_tag.read?release/any(r:r/should_be_running_on__device/any(d:d/${matchesActor}) or r/${belongsToApplicationOrIsManagedByDevice})`,
-	`resin.device_environment_variable.get?device/any(d:d/belongs_to__application/any(a:a/depends_on__application/any(da:da/${ownsDevice})))`,
-	`resin.device_environment_variable.all?device/any(d:d/${matchesActor})`,
-	`resin.application_environment_variable.get?${applicationControlsDevice}`,
+	`resin.application_config_variable.read?application/canAccess()`,
+	`resin.release.read?should_be_running_on__device/canAccess() or ${belongsToApplicationOrIsManagedByDevice}`,
```
Comment (afitzek) [inline src/lib/auth.ts]: `${belongsToApplicationOrIsManagedByDevice}` could become `belongs_to__application/canAccess()` I think?

Comment (Page-) [reply src/lib/auth.ts]: Done, good spot - I was checking for reads that didn't already use `canAccess` which missed this one that did.. but only for one half

File: src/lib/auth.ts [hunk 5]
```diff
@@ -58,58 +64,75 @@ export const ROLES: {
...
+	'resin.release_tag.read?release/canAccess()',
+	'resin.device_environment_variable.read?device/canAccess()',
+	...writePerms(
+		'resin.device_environment_variable',
+		`device/any(d:d/${matchesActor})`,
+	),
+	'resin.application_environment_variable.read?application/canAccess()',
 
 	// Dependent device permissions
+	'resin.device.read?belongs_to__application/any(a:a/depends_on__application/canAccess())',
```
Comment (Page-) [inline src/lib/auth.ts]: Updated this as well, @afitzek just to double check it's fine to have a nested `canAccess` like this right?

Comment (afitzek) [reply src/lib/auth.ts]: I think this should work as intended: You can read the device resource, if there exists a `belongs_to__application` that has a `depends_on__application` that is readable by the device.

**Pull request merged**